renku/core/metadata/database.py

```diff
@@ -751,7 +751,7 @@ def _deserialize_helper(self, data, create=True):
                 if "@reference" in data and data["@reference"]:  # A reference
                     assert create, f"Cannot deserialize a reference without creating an instance {data}"
                     new_object = self._database.get_cached(oid)
-                    if new_object:
+                    if new_object is not None:
```

Comment:
just out of curiosity: what's the difference? between `if obj` and `if obj is not None`, apart from semantics? (sorry for the noise/not being fluent enough in python)